src/graphql/schema.test.js

```diff
@@ -152,6 +152,24 @@ describe("graphql/schema", () => {
         'field "O"/"foo" has invalid type "Color" of kind "ENUM"'
       );
     });
+    it("disallows objects with node fields of unfaithful type", () => {
+      const s = Schema;
+
+      expect(() => {
+        s.node("Color", s.unfaithful(["Color", "Color"]));
+      }).toThrowError("duplicate unfaithful typename Color");
```

Comment:
>that the map is injective modulo permutation of inputs

Can you elaborate on this?

Reply:
Elaborated here:
<https://github.com/sourcecred/sourcecred/pull/1095#pullrequestreview-206666987>

(GitHub doesn’t let you reply to an inline comment on a commit that has
been force-pushed if you have an pending review. This behavior is
actually kind of my fault, because it’s their somewhat hacky fix to a
nasty bug that I reported. But whatever.)